fastlane/swift/MainProcess.swift

```diff
@@ -0,0 +1,58 @@
+// MainProcess.swift
+// Copyright (c) 2020 FastlaneTools
+
+//
+//  ** NOTE **
+//  This file is provided by fastlane and WILL be overwritten in future updates
+//  If you want to add extra functionality to this project, create a new file in a
+//  new group so that it won't be marked for upgrade
+//
+
+import Foundation
+
+let argumentProcessor = ArgumentProcessor(args: CommandLine.arguments)
+let timeout = argumentProcessor.commandTimeout
+
+class MainProcess {
+    var doneRunningLane = false
```

Comment:
Question there! How is this different than `fastlane/swift/main.swift`? It looks similar but I’m not able to tell how this gets used differently than the other one 😇

Reply:
So the issue is that Xcode generates an executable when there's a main.swift, which is considered entry-point for executables, but SPM does the same, uses main.swift as the executable entry point, which conflicts with the fact that Fastlane is not distributed as a executable but as a library. The main.swift lives in the package which declares itself as a executable. That's the reason of the duplication.